libdd-profiling/src/internal/profile/mod.rs

```diff
@@ -297,6 +393,29 @@ impl Profile {
             Self::backup_period(period),
             Self::backup_sample_types(sample_types),
             None,
+            None,
+        )
+    }
+
+    /// Tries to create a profile with the given period and sample types. The
+    /// [`StringId2`]s should belong to the provided [`ProfilesDictionary`].
+    #[inline(never)]
+    #[cold]
+    pub fn try_new2(
+        profiles_dictionary: crate::profiles::collections::Arc<ProfilesDictionary>,
+        sample_types: &[ValueType2],
+        period: Option<Period2>,
+    ) -> io::Result<Self> {
```

Comment:
Having tried this out in PHP, I've decided that it's too much churn in the code to do this at this time. I'm going to adjust the signature so that it takes ValueType and Period rather than ValueType2 and Period2 (and remove the `*2` types in these cases). I will keep the `profiles_dictionary` argument but rename the method, maybe `try_new_with_dictionary` or something.